pkg/cloudprovider/vsphere/nodemanager_test.go

```diff
@@ -267,3 +268,825 @@ func TestReturnIPsFromSpecificFamily(t *testing.T) {
+					},
+				},
+				networks: []vimtypes.GuestNicInfo{
+					{
+						Network: "internal_net",
+						IpAddress: []string{
+							"fe80::3",
+							"fd00:cccc::1",
+							"fd00:cccc::2",
+						},
+					},
+					{
+						Network: "external_net",
+						IpAddress: []string{
+							"fe80::2",
+							"fd00:bbbb::1",
+							"fd00:bbbb::2",
+						},
+					},
+				},
+			},
+			expectedIPs: []v1.NodeAddress{
+				{Type: "InternalIP", Address: "fd00:cccc::1"},
+				{Type: "ExternalIP", Address: "fd00:bbbb::1"},
+			},
+		},
+		{
+			testName: "ByDefaultSelectionIPv6",
+			setup: testSetup{
+				ipFamilyPriority: []string{"ipv6"},
+				cpiConfig:        nil,
+				networks: []vimtypes.GuestNicInfo{
+					{
+						Network: "test_k8s_tenant_c123",
+						IpAddress: []string{
+							"fe80::3",
+							"fd00:cccc::1",
+							"fd00:cccc::2",
+						},
+					},
+					{
+						Network: "test_another_nic",
+						IpAddress: []string{
+							"fe80::2",
+							"fd00:bbbb::1",
+							"fd00:bbbb::2",
+						},
+					},
+				},
+			},
+			expectedIPs: []v1.NodeAddress{
+				{Type: "InternalIP", Address: "fd00:cccc::1"},
+				{Type: "ExternalIP", Address: "fd00:cccc::1"},
+			},
+		},
+		{
+			testName: "ByNetworkNameAndTwoNICs_desiredIPsAfterFirstNIC",
+			setup: testSetup{
+				ipFamilyPriority: []string{"ipv4"},
+				cpiConfig: &ccfg.CPIConfig{
+					Nodes: ccfg.Nodes{
+						InternalVMNetworkName: "internal_net",
+						ExternalVMNetworkName: "external_net",
+					},
+				},
+				networks: []vimtypes.GuestNicInfo{
+					{
+						Network: "test_k8s_tenant_c123",
+						IpAddress: []string{
+							"127.0.0.6",
+							"169.0.1.2",
+						},
+					},
+					{
+						Network: "internal_net",
+						IpAddress: []string{
+							"10.10.10.10",
+						},
+					},
+					{
+						Network: "external_net",
+						IpAddress: []string{
+							"172.15.108.11",
+						},
+					},
+				},
+			},
+			expectedIPs: []v1.NodeAddress{
+				{Type: "InternalIP", Address: "10.10.10.10"},
+				{Type: "ExternalIP", Address: "172.15.108.11"},
+			},
+		},
+		{
+			// This seems like undesireable behavior.
+			testName: "BySubnetAndTwoNICs_desiredIPsAfterFirstNIC",
+			setup: testSetup{
+				ipFamilyPriority: []string{"ipv4"},
+				cpiConfig: &ccfg.CPIConfig{
+					Nodes: ccfg.Nodes{
+						InternalNetworkSubnetCIDR: "10.10.0.0/16",
+						ExternalNetworkSubnetCIDR: "172.15.0.0/16",
+					},
+				},
+				networks: []vimtypes.GuestNicInfo{
+					{
+						Network: "test_k8s_tenant_c123",
+						IpAddress: []string{
+							"127.0.0.6",
+							"169.0.1.2",
+						},
+					},
+					{
+						Network: "internal_net",
+						IpAddress: []string{
+							"10.10.1.22",
+						},
+					},
+					{
+						Network: "external_net",
+						IpAddress: []string{
+							"172.15.108.11",
+						},
+					},
+				},
+			},
+			expectedIPs: []v1.NodeAddress{
+				{Type: "InternalIP", Address: "169.0.1.2"},
+				{Type: "ExternalIP", Address: "169.0.1.2"},
```

Comment:
This scenario is surprising to us, it seems the desired IPs are not selected, and instead default selection happens.

Reply:
This seems like a bug to be honest :thinking:

Reply:
Hooray for tests that catch bugs :)

Reply:
yeah +1 - I believe this is a bug